app/src/main/java/com/gh4a/job/NotificationsJob.java

```diff
@@ -0,0 +1,129 @@
+                        TimeUnit.MINUTES.toMillis(5))
+                .setRequiredNetworkType(JobRequest.NetworkType.CONNECTED)
+                .setRequirementsEnforced(true)
+                .setUpdateCurrent(true)
+                .build()
+                .schedule();
+    }
+
+    public static void cancelJob() {
+        JobManager.instance().cancelAllForTag(NotificationsJob.TAG);
+    }
+
+    @NonNull
+    @Override
+    protected Result onRunJob(Params params) {
+        try {
+            List<Notification> notifications =
+                    NotificationListLoader.loadNotifications(false, false);
+
+            if (notifications.isEmpty()) {
+                return Result.SUCCESS;
+            }
+
+            NotificationManagerCompat notificationManager =
+                    NotificationManagerCompat.from(getContext());
+            notificationManager.cancelAll();
+
+            int accentColor = UiUtils.resolveColor(getContext(), R.attr.colorAccent);
+
+            showSummaryNotification(notificationManager, notifications.size(), accentColor);
+            for (Notification notification : notifications) {
+                showSingleNotification(notificationManager, accentColor, notification);
+            }
+        } catch (IOException e) {
+            return Result.FAILURE;
+        }
+
+        return Result.SUCCESS;
+    }
+
+    private void showSingleNotification(NotificationManagerCompat notificationManager,
+            int accentColor, Notification notification) {
+        Repository repository = notification.getRepository();
+        String title = repository.getOwner().getLogin() + "/" + repository.getName();
+        long when = notification.getUpdatedAt() != null
+                ? notification.getUpdatedAt().getTime()
+                : System.currentTimeMillis();
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(getContext(),
+                Gh4Application.CHANNEL_GITHUB_NOTIFICATIONS)
+                .setSmallIcon(R.drawable.octodroid)
+                .setGroup(GROUP_ID_GITHUB)
+                .setWhen(when)
+                .setShowWhen(true)
+                .setColor(accentColor)
+                .setContentTitle(title)
+                .setAutoCancel(true)
+                .setContentText(notification.getSubject().getTitle());
+
+        String url = notification.getSubject().getUrl();
+        if (url != null) {
+            Uri uri = ApiHelpers.normalizeUri(Uri.parse(url));
+            Intent intent = BrowseFilter.makeRedirectionIntent(getContext(), uri, null);
+            PendingIntent pendingIntent =
+                    PendingIntent.getActivity(getContext(), 0, intent, 0);
+            builder.setContentIntent(pendingIntent);
+        }
+
+        int id = ID_SUMMARY + 1;
+        try {
+            id = Integer.parseInt(notification.getId());
+        } catch (NumberFormatException e) {
+            // ignored
```

Comment:
If this fails (it probably won't, though) we'll nuke another notification. Suggestion: Drop the ID parsing and use the offset into the notification array (+ 1 for summary) as notification ID.